solr/solrj/src/java/org/apache/solr/common/SolrDocument.java

```diff
@@ -373,6 +374,23 @@ public int size() {
     return _fields.values();
   }
 
+  public Map<String, Object> getChildDocumentsMap() {
```

Comment:
This method is tempting; it's unsurprising you added it, but I'd rather we not have it.  It's not particularly cheap; raises questions of mutability, and it's really not necessary; only one caller (for indexing, recunwrap) needs this algorithm; and it need not build up some HashMap as we see here to do its job.